gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -801,12 +813,14 @@ public void flush(String dbName, String tableName) throws IOException {
         Transaction transaction = tableMetadata.transaction.get();
         Map<String, String> props = tableMetadata.newProperties.or(
             Maps.newHashMap(tableMetadata.lastProperties.or(getIcebergTable(tid).properties())));
-        String topic = props.get(TOPIC_NAME_KEY);
+        //Set data offset range
+        setDatasetOffsetRange(tableMetadata, props);
+        String topicName = getTopicName(tid, tableMetadata);
```

Comment:
Is tableMetadataMap.get(tid).setDatasetName(gmce.getDatasetIdentifier().getNativeName()); required after this change to a new way to get topic name?

Reply:
We are still fallback to the previous topic name calculated from NativeName here in case datasetOffsetRange does not exist.  Also GTE require this datasetName to set the HDFS dataset path information